### PR TITLE
[T280223] Add readme for each docker image we publish

### PR DESCRIPTION
**Pull request by toban**

### CODE REVIEW

File: Docker/build/Wikibase/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## wikibase docker image
```
Comment (manicki) [inline Docker/build/Wikibase/README.md:1]:
```suggestion
## Wikibase Docker image
```

File: Docker/build/QuickStatements/README.md [hunk 2]
```diff
@@ -4,14 +4,20 @@ Quickstatements2 as seen at  [https://github.com/magnusmanske/quickstatements](h
...
-`WIKIBASE_SCHEME_AND_HOST` | NONE     | Host and port of wikibase instance as seen by QS container                                                  
-`WB_PUBLIC_HOST_AND_PORT`  | NONE     | Host and port of wikibase as seen by the user's browser
-`QS_PUBLIC_HOST_AND_PORT`  | NONE     | Host and port of QS as seen by the user's browser      
-`OAUTH_CONSUMER_KEY`       | NONE     | OAuth consumer key (obtained from wikibase)            
-`OAUTH_CONSUMER_SECRET`    | NONE     | OAuth consumer key (obtained from wikibase)            
-`PHP_TIMEZONE`             | UTC      | setting of php.ini date.timezone                       
+Variable                             | Default      | Description                                            
+-------------------------------------|--------------|------------
+`WIKIBASE_SCHEME_AND_HOST`           | NONE         | Host and port of wikibase instance as seen by QS
+`WB_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of wikibase as seen by the user's browser
```
Comment (manicki) [inline Docker/build/QuickStatements/README.md:10]:
```suggestion
`WB_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of Wikibase as seen by the user's browser
```

File: Docker/build/QuickStatements/README.md [hunk 3]
```diff
@@ -4,14 +4,20 @@ Quickstatements2 as seen at  [https://github.com/magnusmanske/quickstatements](h
...
+`QS_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of QS as seen by the user's browser      
+`OAUTH_CONSUMER_KEY`                 | NONE         | OAuth consumer key (obtained from wikibase)
```
Comment (manicki) [inline Docker/build/QuickStatements/README.md:12]:
```suggestion
`OAUTH_CONSUMER_KEY`                 | NONE         | OAuth consumer key (obtained from Wikibase)            
```

File: Docker/build/QuickStatements/README.md [hunk 4]
```diff
@@ -4,14 +4,20 @@ Quickstatements2 as seen at  [https://github.com/magnusmanske/quickstatements](h
...
-`OAUTH_CONSUMER_KEY`       | NONE     | OAuth consumer key (obtained from wikibase)            
-`OAUTH_CONSUMER_SECRET`    | NONE     | OAuth consumer key (obtained from wikibase)            
-`PHP_TIMEZONE`             | UTC      | setting of php.ini date.timezone                       
+Variable                             | Default      | Description                                            
+-------------------------------------|--------------|------------
+`WIKIBASE_SCHEME_AND_HOST`           | NONE         | Host and port of wikibase instance as seen by QS
+`WB_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of wikibase as seen by the user's browser
+`QS_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of QS as seen by the user's browser      
+`OAUTH_CONSUMER_KEY`                 | NONE         | OAuth consumer key (obtained from wikibase)            
+`OAUTH_CONSUMER_SECRET`              | NONE         | OAuth consumer key (obtained from wikibase)
```
Comment (manicki) [inline Docker/build/QuickStatements/README.md:13]: is this also consumer key (see line above)?

File: Docker/build/QuickStatements/README.md [hunk 5]
```diff
@@ -4,14 +4,20 @@ Quickstatements2 as seen at  [https://github.com/magnusmanske/quickstatements](h
...
-`WB_PUBLIC_HOST_AND_PORT`  | NONE     | Host and port of wikibase as seen by the user's browser
-`QS_PUBLIC_HOST_AND_PORT`  | NONE     | Host and port of QS as seen by the user's browser      
-`OAUTH_CONSUMER_KEY`       | NONE     | OAuth consumer key (obtained from wikibase)            
-`OAUTH_CONSUMER_SECRET`    | NONE     | OAuth consumer key (obtained from wikibase)            
-`PHP_TIMEZONE`             | UTC      | setting of php.ini date.timezone                       
+Variable                             | Default      | Description                                            
+-------------------------------------|--------------|------------
+`WIKIBASE_SCHEME_AND_HOST`           | NONE         | Host and port of wikibase instance as seen by QS
+`WB_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of wikibase as seen by the user's browser
+`QS_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of QS as seen by the user's browser
```
Comment (manicki) [inline Docker/build/QuickStatements/README.md:11]:
```suggestion
`QS_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of Quick Statements as seen by the user's browser      
```

File: Docker/build/Wikibase/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+Directory                         | Description
+--------------------------------- | ------------------------------------------------------------------------------
+`/var/www/html`                   | Base Mediawiki directory
+`/var/www/html/skins`             | Mediawiki skins directory
+`/var/www/html/extensions`        | Mediawiki extensions directory
+
+File                              | Description
+--------------------------------- | ------------------------------------------------------------------------------
+`/LocalSettings.php.template`     | Template for Mediawiki Localsettings.php (substituted to `/var/www/html/LocalSettings.php` at runtime)
```
Comment (manicki) [inline Docker/build/Wikibase/README.md:33]:
```suggestion
`/LocalSettings.php.template`     | Template for Mediawiki LocalSettings.php (substituted to `/var/www/html/LocalSettings.php` at runtime)
```

File: Docker/build/WikibaseBundle/README.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
+## wikibase bundle docker image
```
Comment (manicki) [inline Docker/build/WikibaseBundle/README.md:1]:
```suggestion
## Wikibase Bundle Docker image
```

File: Docker/build/WikibaseBundle/README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+`/var/www/html`                       | Base Mediawiki directory
+`/var/www/html/skins`                 | Mediawiki skins directory
+`/var/www/html/extensions`            | Mediawiki extensions directory
+`/var/www/html/LocalSettings.d`       | Bundle extension configuration directory
+`/extra-install`                      | Extra install scripts for automatic setup
+`/templates/`                         | Directory containing templates
+
+File                                  | Description
+--------------------------------------|-------------------------------------------------------------------------------
+`/LocalSettings.php.template`         | Template for Mediawiki Localsettings.php (substituted to `/var/www/html/LocalSettings.php` at runtime)
```
Comment (manicki) [inline Docker/build/WikibaseBundle/README.md:37]:
```suggestion
`/LocalSettings.php.template`         | Template for Mediawiki LocalSettings.php (substituted to `/var/www/html/LocalSettings.php` at runtime)
```

File: Docker/build/WikibaseBundle/README.md [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+`/var/www/html/LocalSettings.php`     | LocalSettings.php location, when passed in `/LocalSettings.php.template` will not be used. install.php & update.php will also not be run.
+`/extra-install/ElasticSearch.sh`     | Script for automatically creating elasticsearch indicies
```
Comment (manicki) [inline Docker/build/WikibaseBundle/README.md:39]:
```suggestion
`/extra-install/ElasticSearch.sh`     | Script for automatically creating Elasticsearch indices
```

File: Docker/build/WikibaseBundle/README.md [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+`/extra-install/QuickStatements.sh`   | Script for automatically creating Oauth consumer for quickstatements
```
Comment (manicki) [inline Docker/build/WikibaseBundle/README.md:40]:
```suggestion
`/extra-install/QuickStatements.sh`   | Script for automatically creating OAuth consumer for Quick Statements
```

File: Docker/build/QueryService/README.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+File                              | Description
+--------------------------------- | ------------------------------------------------------------------------------
+`/wdqs/whitelist.txt`             | Whitelist file for other SPARQL endpoints
+`/wdqs/RWStore.properties`        | Properties for the service
+`/templates/mwservices.json`      | Template for MediaWiki services (substituted to `/wdqs/mwservices.json` at runtime)
+
+### Troubleshooting
+
+* The query service is not running or seems to get killed by the OS?
+* The image requires more than 2GB of available RAM to start. While being developed the dev machine had 4GB of RAM.
```
[the remark this answers is not in the document]
Comment (toban) [reply Docker/build/QueryService/README.md:41]: Seems to be an old comment from wikibase-docker i cannot attest to. Our dev/test system is on 8 gb

File: Docker/build/QueryService/README.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+When creating a new image RWStore.properties will need to be updated to match the properties used in production at the time the snapshot was being used.
+
+For this reason it is easier to only create new releases for wdqs versions currently being used in Wikidata production.
+
+When creating a new release the WMF Search platform team should be contacted for help syncing the wdqs version and RWStore.properties file.
+
+The process is generally:
+ - Create a new directory using a previous one as an example
+ - Update the service snapshot that is being fetched
+ - Update the CI build by checking the steps in the main README Development section in this repo.
```
[the remark this answers is not in the document]
Comment (toban) [reply Docker/build/QueryService/README.md:58]: yeah lets get rid of this

File: Docker/build/QuickStatements/README.md (hunk 4 above)
Comment (toban) [reply Docker/build/QuickStatements/README.md:13]: Looks like some old blunder, good catch

**Review by manicki — APPROVED** [review]
Thanks for gazillion fixes.

File: Docker/build/QuickStatements/README.md [hunk 13]
```diff
@@ -1,22 +1,28 @@
...
+Variable                             | Default      | Description                                            
+-------------------------------------|--------------|------------
+`WIKIBASE_SCHEME_AND_HOST`           | NONE         | Host and port of Wikibase instance as seen by Quick Statements
+`WB_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of Wikibase as seen by the user's browser
+`QS_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of Quick Statements as seen by the user's browser      
+`OAUTH_CONSUMER_KEY`                 | NONE         | OAuth consumer key (obtained from Wikibase)            
+`OAUTH_CONSUMER_SECRET`              | NONE         | OAuth consumer secret (obtained from wikibase)
+`PHP_TIMEZONE`                       | UTC          | setting of php.ini date.timezone
+`MW_SITE_LANG`                       | NONE         | Site language
+`MW_SITE_NAME`                       | NONE         | Site name
```
Comment (mariushoch) [inline Docker/build/QuickStatements/README.md:16]: Default is `wikibase-docker` according to `Docker/build/QuickStatements/Dockerfile`.

File: Docker/build/QuickStatements/README.md [hunk 14]
```diff
@@ -1,22 +1,28 @@
...
-`PHP_TIMEZONE`             | UTC      | setting of php.ini date.timezone                       
+Variable                             | Default      | Description                                            
+-------------------------------------|--------------|------------
+`WIKIBASE_SCHEME_AND_HOST`           | NONE         | Host and port of Wikibase instance as seen by Quick Statements
+`WB_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of Wikibase as seen by the user's browser
+`QS_PUBLIC_SCHEME_HOST_AND_PORT`     | NONE         | Host and port of Quick Statements as seen by the user's browser      
+`OAUTH_CONSUMER_KEY`                 | NONE         | OAuth consumer key (obtained from Wikibase)            
+`OAUTH_CONSUMER_SECRET`              | NONE         | OAuth consumer secret (obtained from wikibase)
+`PHP_TIMEZONE`                       | UTC          | setting of php.ini date.timezone
+`MW_SITE_LANG`                       | NONE         | Site language
```
[the remark this answers is not in the document]
Comment (toban) [reply Docker/build/QuickStatements/README.md:15]: good catch

File: Docker/build/QuickStatements/README.md (hunk 13 above)
Comment (toban) [reply Docker/build/QuickStatements/README.md:16]: good catch

File: Docker/build/QueryService/README.md [hunk 15]
```diff
@@ -0,0 +1,50 @@
+## WDQS Docker image
+
+Wikibase specific Blazegraph image.
+
+### Upgrading
+
+When upgrading between WDQS versions the data stored in `/wdqs/data` may not be compatible with the newer version.
+When testing the new image if no data appears to be loaded into the Query Service you will need to reload the data.
+
+If all changes are still in RecentChanges then simply removing `/wdqs/data` and restarting the service should reload all data.
```
[the remark this answers is not in the document]
Comment (toban) [reply Docker/build/QueryService/README.md:10]: Do you mean that we should describe further how WDQS is updating its data or what RecentChanges are?

File: Docker/build/QueryService/README.md [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+If you can not use RecentChanges then you will need to reload from an RDF dump:
+ - Make an RDF dump from your Wikibase instance: https://github.com/wikimedia/mediawiki-extensions-Wikibase/blob/master/repo/maintenance/dumpRdf.php
```
[the remark this answers is not in the document]
Comment (toban) [reply Docker/build/QueryService/README.md:13]: I found this https://www.mediawiki.org/wiki/Wikibase/Indexing/RDF_Dump_Format that describes the format a bit but there seems to be lacking a guide for how to use `dumpRdf.php` indeed. I suggest we create a ticket to make such a page maybe on mediawiki.org/Wikibase? How does that sound?

File: Docker/build/QueryService/README.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (mariushoch) [reply Docker/build/QueryService/README.md:10]: What RecentChanges are (and how one can find out that the changes are still in RC).

[the remark this answers is not in the document]
Comment (toban) [reply Docker/build/QueryService/README.md]: Right, added some links to the mediawiki documentation and rephrased it a bit.

File: Docker/build/QueryService/README.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (mariushoch) [reply Docker/build/QueryService/README.md]: Sounds like a good idea… let's create a ticket and make a note there to link the doc in this README once it's written.

File: Docker/build/QueryService/README.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (tarrow) [reply Docker/build/QueryService/README.md]: Looks decent enough to me now

**Pull request merged**